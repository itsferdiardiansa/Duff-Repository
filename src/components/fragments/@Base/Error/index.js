/* eslint-disable no-unused-vars */
// import { defineComponent, getCurrentInstance, h } from "vue";

export { default } from './Error'
// import ErrorComponent from './Error'

// export default defineComponent({
//   render() {
//     const root = getCurrentInstance()
//     console.log(root)

//     return h(ErrorComponent, {})
//   }
// })