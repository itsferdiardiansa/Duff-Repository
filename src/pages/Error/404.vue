<template>
  <div class="w-full flex flex-col items-center justify-center -mt-20">
    <h3 class="text-3xl">Ooops, Page not found :(</h3>

    <div class="img h-2/3">
      <img svg-inline class="h-full pl-20" src="@icon/error-page.svg" />
    </div>

    <div class="btn">
      <Button label="Back to Home" variant="primary" @click="redirectToHome" />
    </div>
  </div>
</template>
<script>
import { useRouter } from 'vue-router'
import Button from '@common/Button'

export default {
  components: {
    Button,
  },
  setup() {
    const router = useRouter()

    const redirectToHome = () => {
      router.push('/')
    }

    return {
      redirectToHome,
    }
  },
}
</script>
