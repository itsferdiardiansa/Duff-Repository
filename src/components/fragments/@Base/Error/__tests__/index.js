import { mount } from '@vue/test-utils';
import Error from '@base/Error';

describe('fragment/Base/Error', () => {
  it('will render and match snapshot', () => {
    const Component = mount(<Error />);

    expect(Component.html()).toMatchSnapshot();
  });

  it('will trigger event and reload the oage', async () => {
    delete global.window.location;
    global.window = Object.create(window);
    global.window.location = {
      reload: jest.fn(),
    };

    const Component = mount(<Error />);

    await Component.find('button').trigger('click');
    expect(window.location.reload).toHaveBeenCalled();
  });
});
