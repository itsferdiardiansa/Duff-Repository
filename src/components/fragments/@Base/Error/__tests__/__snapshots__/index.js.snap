// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment/Base/Error will render and match snapshot 1`] = `
<div class="ss-error fade-slide-enter-from fade-slide-enter-active">
  <div class="ss-error--img"><img svg-inline=""></div>
  <div class="ss-error--description">
    <div class="message">
      <h1>Sorry there is a problem with our system</h1>
      <p> Wait a minute, let me clean up. Please try again or come back later. </p>
    </div>
    <div class="action-button"><button class="ss-button ss-button--dark ss-button--xl text-bold">
        <div class="ss-button--wrapper">
          <!--v-if-->
          <div class="ss-button--label">Reload</div>
        </div>
      </button></div>
  </div>
</div>
`;
